hermes-consumers/src/main/java/pl/allegro/tech/hermes/consumers/consumer/sender/http/OAuth2AuthProvider.java

```diff
@@ -0,0 +1,65 @@
+package pl.allegro.tech.hermes.consumers.consumer.sender.http;
+
+import pl.allegro.tech.hermes.api.OAuth2AuthenticationData;
+import pl.allegro.tech.hermes.api.Subscription;
+import pl.allegro.tech.hermes.common.cache.zookeeper.NodeCache;
+import pl.allegro.tech.hermes.consumers.consumer.ConsumerMessageSender;
+import pl.allegro.tech.hermes.consumers.consumer.Message;
+
+import com.github.scribejava.core.builder.api.DefaultApi20;
+
+import com.github.scribejava.core.builder.ServiceBuilder;
+import com.github.scribejava.core.model.OAuth2AccessToken;
+import com.github.scribejava.core.oauth.OAuth20Service;
+import com.github.scribejava.core.model.OAuthConfig;
+
+public class OAuth2AuthProvider implements HttpAuthorizationProvider {
```

Comment:
Consumer is single-threaded and is responsible for reading message and sending payload without waiting for response. Anything that goes between reading message and sending payload can be blocking. There is not much to do until token is authorised. We could have separate connection pool per OAuth gateway though. Not sure what is going on in ScribeJava. We have to read what that library is doing before merging this.